packages/fuels-core/src/encoding_utils.rs

```diff
@@ -0,0 +1,105 @@
+use crate::{ParamType, WORD_SIZE};
+
+/// If given a non-empty slice it will return the number of `WORD`s needed to fit the biggest of the provided types.
+pub fn max_by_encoding_width(params: &[ParamType]) -> Option<usize> {
```

Comment:
I wanted to call it that, but there is one thing that rubs me wrong about the name:
It might be misunderstood to mean that we're trying to encode the whole slice -- all of the types present in it.

That is why I tried to communicate that we're somehow judging them based on their encoding size, and `max_by_encoding_width` was the best I could come up with.

But I just thought of something as I was writing this. Please see how you like the new version :)

Reply:
Changed the name as per the other comment. Now it deals with enums and is called
`compute_encoding_width_of_enum`